packages/react/src/FilteredActionList/FilteredActionListWithModernActionList.tsx

```diff
@@ -123,44 +165,86 @@ export function FilteredActionList({
     [onListContainerRefChanged],
   )
 
-  useEffect(() => {
-    onInputRefChanged?.(inputRef)
-  }, [inputRef, onInputRefChanged])
-
+  // Only use focus zone when the new feature flag is disabled (old behavior)
   useFocusZone(
-    {
-      containerRef: {current: listContainerElement},
-      bindKeys: FocusKeys.ArrowVertical | FocusKeys.PageUpDown,
-      focusOutBehavior: 'wrap',
-      focusableElementFilter: element => {
-        return !(element instanceof HTMLInputElement)
-      },
-      activeDescendantFocus: inputRef,
-      onActiveDescendantChanged: (current, previous, directlyActivated) => {
-        activeDescendantRef.current = current
-
-        if (current && scrollContainerRef.current && directlyActivated) {
-          scrollIntoView(current, scrollContainerRef.current, menuScrollMargins)
+    !usingRemoveActiveDescendant
+      ? {
+          containerRef: {current: listContainerElement},
+          bindKeys: FocusKeys.ArrowVertical | FocusKeys.PageUpDown,
+          focusOutBehavior: 'wrap',
+          focusableElementFilter: element => {
+            return !(element instanceof HTMLInputElement)
+          },
+          activeDescendantFocus: inputRef,
+          onActiveDescendantChanged: (current, previous, directlyActivated) => {
+            activeDescendantRef.current = current
+
+            if (current && scrollContainerRef.current && directlyActivated) {
+              scrollIntoView(current, scrollContainerRef.current, menuScrollMargins)
+            }
+          },
         }
-      },
-    },
+      : undefined,
     [
       // List container isn't in the DOM while loading.  Need to re-bind focus zone when it changes.
       listContainerElement,
+      usingRemoveActiveDescendant,
     ],
   )
 
+  // TODO remove with useRemoveActiveDescendant
+
+  useEffect(() => {
+    onInputRefChanged?.(inputRef)
+  }, [inputRef, onInputRefChanged])
+
   useEffect(() => {
-    // if items changed, we want to instantly move active descendant into view
-    if (activeDescendantRef.current && scrollContainerRef.current) {
-      scrollIntoView(activeDescendantRef.current, scrollContainerRef.current, {
-        ...menuScrollMargins,
-        behavior: 'auto',
-      })
```

Comment:
is this not needed anymore when the FF is off?

Reply:
this was a amiss. Added it back!